File: backend/Responses/BaseResponse.cs
using backend.Models;

namespace backend.Responses;

public class BaseResponse
{
    public string Message { get; }
    public RequestVariant Variant { get; }

    public BaseResponse(string message, RequestVariant variant)
    {
        this.Message = message;
        this.Variant = variant;
    }

    public virtual object ToObject()
    {
        return new
        {
            message = this.Message,
            variant = this.Variant.ToString(),
        };
    }
}

File: web/src/responses/base-response.ts
import { AlertVariantEnum } from "@/enums";

export interface BaseResponseProps {
  message: string;
  variant: AlertVariantEnum;
}

export class BaseResponse {
  public message: string;
  public variant: AlertVariantEnum;

  constructor({ message, variant }: BaseResponseProps) {
    this.message = message;
    this.variant = variant;
  }
}